simespresso/io_QE/espresso_class.py

```diff
@@ -0,0 +1,815 @@
+                f.write(line)
+                for i in range(0, n_elements[0]):
+                    x = i * base_vector[0]
+                    for j in range(0, n_elements[1]):
+                        y = j * base_vector[0]
+                        for k in range(0, n_elements[2]):
+                            z = k * base_vector[0]
+                            density = xyz_array[i, j, k]
+                            line = 'C ' + str(x) + ' ' + str(y) + ' ' + str(z) + \
+                                   ' ' + str(density) + '\n'
+                            f.write(line)
+            except:
+                print('error writing aviz file')
+        return
+
+    def read_xyz(self, n_latticepoints, file_iter):
+        line = file_iter.next()
+        x_points = n_latticepoints[0]
+        y_points = n_latticepoints[1]
+        z_points = n_latticepoints[2]
+        x_count = 0
+        y_count = 0
+        z_count = 0
+        charge_density = np.zeros([x_points, y_points, z_points])
+        line_number = 0
+        try:
+            while line is not None:
+                line_number = line_number + 1
+                print('x{0} y{1} z{2} line:{3}'
+                      .format(x_count, y_count, z_count, str(line)))
+                values = line.split()
+                charges = [float(val) for val in values]
+                #            print('charges:'+str(charges))
+
+                for charge in charges:
+                    charge_density[x_count, y_count, z_count] = charge
+                    x_count += 1
+                    if x_count == x_points:
+                        x_count = 0
+                        y_count += 1
+                        if y_count == y_points:
+                            y_count = 0
+                            z_count += 1
+                            if z_count == z_points:
+                                break
+
+                    line = file_iter.next()
+
+        except StopIteration:
+            print('EOF')
+        if x_count < x_points or y_count < y_points or z_count < z_points:
+            logging.debug('Got fewer points than expected:read {0} of {1} x, '
+                          '{2} of {3} y, {4} of {5} z'.
+                          format(x_count, x_points, y_count,
+                                 y_points, z_count, z_points))
+        return charge_density
+
+    def write_espresso_input_file(self, file_name):
+        """
+        :param file_name: name of the input file to write
+        :return:
+        """
+        SP = self.SP
+        pc = self.pc
+        # write parameters for a particular working input file
+
+        print('attempting to write ' + file_name)
+        try:
+            with open(file_name, 'w') as f:
+                # CONTROL section
+                # apparently a comma is not required at the end of every line
+                line = '&CONTROL\n'
+                f.write(line)
+                if hasattr(self, 'calculation_type'):
```

Comment:
> I would name these attributes from self.calculation_type and self.restart_mode to self._calculation_type and self._restart_mode

This is just a suggestion if these instance variables are not part of the public interface.  You could alternatively just keep them as they are.